### PR TITLE
[Do not merge] remove threadid buffer for RNTuple

### LINKED ISSUE #251: Purge `threadid()`-based buffer from code base
Ever since Julia had task migration between OS threads, this practice has become technically a bug, we need to purge them.

Current idea is to use a `LRU` in place of the threaded() based buffer.

### PR DESCRIPTION
**Pull request by Moelf**
fix #251 

## original performance
```julia
Test Summary:                 | Pass  Total  Time
RNTuple Anchodr/Header/Footer |   12     12  2.5s
Test Summary:          | Pass  Total  Time
RNTuple Schema Parsing |   14     14  0.0s
Test Summary:         | Pass  Total  Time
RNTuple Int32 reading |    3      3  0.9s
Test Summary:             | Pass  Total  Time
RNTuple bit(bool) reading |    1      1  0.3s
Test Summary:                 | Pass  Total  Time
RNTuple std:: container types |   19     19  6.5s
Test Summary:      | Pass  Total  Time
RNTupleCardinality |    1      1  1.5s
Test Summary:          | Pass  Total  Time
RNTuple Split Encoding |    6      6  1.0s
Test Summary:          |Time
RNTuple Type stability | None  0.0s
Test Summary:           | Pass  Total  Time
RNTuple Multi-threading |    3      3  0.1s
Test Summary:                            | Pass  Total   Time
RNTuple Tables.jl and Arrow integration  |    2      2  44.2s
     Testing UnROOT tests passed
```

### CODE REVIEW

**Comment by Moelf** [comment]
the performance is terrible,

using https://juliaconcurrent.github.io/ConcurrentCollections.jl/dev/#ConcurrentCollections.ConcurrentDict

```julia
Test Summary:                 | Pass  Total  Time
RNTuple Anchodr/Header/Footer |   12     12  2.8s
Test Summary:          | Pass  Total  Time
RNTuple Schema Parsing |   14     14  0.0s
Test Summary:         | Pass  Total  Time
RNTuple Int32 reading |    3      3  5.1s
Test Summary:             | Pass  Total  Time
RNTuple bit(bool) reading |    1      1  0.5s
Test Summary:                 | Pass  Total  Time
RNTuple std:: container types |   19     19  9.1s
Test Summary:      | Pass  Total  Time
RNTupleCardinality |    1      1  2.0s
Test Summary:          | Pass  Total   Time
RNTuple Split Encoding |    6      6  13.2s
Test Summary:          |Time
RNTuple Type stability | None  7.1s
Test Summary:           | Pass  Total  Time
RNTuple Multi-threading |    3      3  2.1s
Test Summary:                            | Pass  Total   Time
RNTuple Tables.jl and Arrow integration  |    2      2  44.1s
     Testing UnROOT tests passed
```

**Comment by Moelf** [comment]
[LRU](https://github.com/JuliaCollections/LRUCache.jl) is even worse

```julia
Test Summary:                 | Pass  Total  Time
RNTuple Anchodr/Header/Footer |   12     12  2.4s
Test Summary:          | Pass  Total  Time
RNTuple Schema Parsing |   14     14  0.0s
Test Summary:         | Pass  Total  Time
RNTuple Int32 reading |    3      3  8.0s
Test Summary:             | Pass  Total  Time
RNTuple bit(bool) reading |    1      1  0.4s
Test Summary:                 | Pass  Total  Time
RNTuple std:: container types |   19     19  8.9s
Test Summary:      | Pass  Total  Time
RNTupleCardinality |    1      1  1.9s
Test Summary:          | Pass  Total   Time
RNTuple Split Encoding |    6      6  24.2s
Test Summary:          | Time
RNTuple Type stability | None  13.9s
Test Summary:           | Pass  Total   Time
RNTuple Multi-threading |    3      3  14.1s
Test Summary:                            | Pass  Total   Time
RNTuple Tables.jl and Arrow integration  |    2      2  44.5s
     Testing UnROOT tests passed
```

File: src/iteration.jl [hunk 1]
```diff
@@ -162,48 +160,34 @@ end
...
+    end
 
-function _get_buffer_range(ba::LazyBranch{T, J, B}, tid::Integer, ::Nothing) where {T,J,B}
-    ba.buffer[tid] = basketarray(ba.f, ba.b, -1)  # -1 indicating recovered basket mechanics
-    # FIXME: this range is probably wrong for jagged data with non-empty offsets
-    (ba.b.fBasketEntry[end] + 1)::Int:ba.b.fEntries::Int
+    return br, data
 end
 
+# function _get_buffer_range(ba::LazyBranch{T, J, B}, tid::Integer, ::Nothing) where {T,J,B}
```
Comment (Moelf) [inline src/iteration.jl:185]: @tamasgal do you remember what is this for? didn't hit in test it seems

Comment (tamasgal) [reply src/iteration.jl:185]: We refactored a few things, not sure about the origins 🙈

Comment (Moelf) [reply src/iteration.jl:187]: https://github.com/JuliaHEP/UnROOT.jl/commit/81007e16fe0253e4d9e6652f1cf41ae853c9061b

it's for recovered basket when `seek_idx` is nothing

**Pull request closed without merging**